hack/update-deps.sh

```diff
@@ -20,6 +20,43 @@ set -o pipefail
 
 source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/library.sh
 
+# Remove two type of invalid symlinks
+# 1. Broken symlinks
+# 2. Symlinks pointing outside of the source tree
+# We use recursion to find the canonical path of a symlink
+# because both `realpath` and `readlink -m` are not available on Mac.
+function remove_invalid_symlink() {
+  cd ${REPO_ROOT_DIR}
+
+  local target_file=$1
+  if [[ ! -e ${target_file} ]] ; then
+    echo "Removing broken symlink:" $1
+    unlink ${REPO_ROOT_DIR}/$1
+    return
+  fi
+
+  cd $(dirname ${target_file})
+  target_file=$(basename ${target_file})
+  # Iterate down a [possible] chain of symlinks
+  while [[ -L ${target_file} ]] ; do
+    target_file=$(readlink ${target_file})
+    cd $(dirname ${target_file})
+    target_file=$(basename ${target_file})
+  done
+
+  # Compute the canonicalized name by finding the physical path
+  # for the directory we're in and appending the target file.
+  local phys_dir=`pwd -P`
+  target_file=${phys_dir}/${target_file}
```

Comment:
Seems like this is producing a double slash in the output, e.g.:
```
Removing broken symlink: vendor//k8s.io/kubernetes/cluster/gce/cos
```

Reply:
Oops I commented on the wrong line, but this line is affected too:
```
Removing symlink outside the source tree:  vendor//github.com/prometheus/procfs/fixtures/26231/exe -> /usr/bin/vim
```

Reply:
While I agree that the messages help during development, I think this is too verbose.